spring-xd-dirt/src/test/java/org/springframework/xd/dirt/security/SingleNodeApplicationWithUsersFileTest.java

```diff
@@ -88,7 +90,716 @@ public void testWrongPasswordFails() throws Exception {
 	@Test
 	public void testAuthenticatedAccessToManagementEndpointSucceeds() throws Exception {
 		springXdResource.getMockMvc()
-				.perform(get("/management/metrics").header("Authorization", basicAuthorizationHeader("alice", "alicepwd")))
+				.perform(
+						get("/management/metrics").header("Authorization",
+								basicAuthorizationHeader("alice", "alicepwd")))
 				.andDo(print())
 				.andExpect(status().isOk());
-	}}
+	}
+
+	// Stream Definitions
+
+	@Test
+	public void testAuthenticatedButUnauthorizedAccessToDefinitionsEndpoint() throws Exception {
+		springXdResource.getMockMvc()
+				.perform(
+						get("/streams/definitions").header("Authorization",
+								basicAuthorizationHeader("cartman", "cartmanpwd")))
+				.andDo(print())
+				.andExpect(status().isForbidden());
+	}
+
+	@Test
+	public void testAuthenticatedButUnauthorizedAccessToDefinitionsEndpointWithXmlExtension() throws Exception {
+		springXdResource.getMockMvc()
+				.perform(
+						get("/streams/definitions.xml").header("Authorization",
```

Comment:
We should use JUnit @Parameterized to cover all 3 cases automatically, or this may quickly go beyond madness.